source/autocomplete/autocomplete.cpp

```diff
@@ -155,4 +156,69 @@ void Autocomplete<type::idx_t>::compute_score(type::PT_Data &pt_data, georef::Ge
     }
 }
 
+template<class T>
+std::vector<typename Autocomplete<T>::fl_quality>
+Autocomplete<T>::compute_vec_quality(const std::string& str ,
+                                     const std::vector<T>& index_result,
+                                     const navitia::georef::GeoRef& geo_ref,
+                                     std::function<bool(T)> keep_element,
+                                     int wordLength) const{
+    std::vector<fl_quality> vec_quality;
+    std::vector<std::string> tokens_req;
+    {
+        auto tmp_str = strip_accents_and_lower(str);
+        boost::split(tokens_req, tmp_str, boost::is_any_of(" "));
+    }
+    for(auto i : index_result){
+        if(keep_element(i)) {
+            fl_quality quality;
+
+            quality.idx = i;
+            quality.nb_found = word_quality_list.at(quality.idx).word_count;
+            quality.word_len = wordLength;
+            quality.score = word_quality_list.at(quality.idx).score;
+            quality.quality = 100;
+
+            // whole_way_name = road name + city name
+            auto whole_way_name = geo_ref.ways[i]->name;
+            for (const auto* admin: geo_ref.ways[i]->admin_list) {
+                if(admin && admin->level == 8) {
+                    whole_way_name += (" " + admin->name);
+                }
+            }
+            whole_way_name = strip_accents_and_lower(whole_way_name);
+            boost::algorithm::replace_all(whole_way_name, "-", " ");
+
+            boost::tokenizer<> tokens_candidate(whole_way_name);
+            auto it_req = std::begin(tokens_req);
+            auto it_can = std::begin(tokens_candidate);
+            for(; it_req != std::end(tokens_req) && it_can != std::end(tokens_candidate); ++it_req, ++it_can) {
+                quality.score += (*it_can).find(*it_req) != std::string::npos;
```

Comment:
not exact match,  if the request word is `sai seb`, the score will have two additional points if the candidate is `saint sebastien`